probes/testsRunInCI/impl_test.go

```diff
@@ -0,0 +1,488 @@
+// Testing only for the outcome is insufficient, because the
+// values of the findings are important to the probe.
+func Test_Run(t *testing.T) {
+	t.Parallel()
+	//nolint:govet
+	tests := []struct {
+		name     string
+		raw      *checker.RawResults
+		findings []*finding.Finding
+		err      error
+	}{
+		{
+			name: "Has 1 CIInfo which has a successful CheckRun.",
+			raw: &checker.RawResults{
+				CITestResults: checker.CITestData{
+					CIInfo: []checker.RevisionCIInfo{
+						{
+							HeadSHA:           "HeadSHA",
+							PullRequestNumber: 1,
+							CheckRuns: []clients.CheckRun{
+								{
+									Status:     "completed",
+									Conclusion: "success",
+									App:        clients.CheckRunApp{Slug: "e2e"},
+								},
+							},
+							Statuses: []clients.Status{
+								{
+									State:     "not successful",
+									Context:   CheckCITests,
+									TargetURL: "e2e",
+								},
+							},
+						},
+					},
+				},
+			},
+			findings: []*finding.Finding{
+				{
+					Outcome:  finding.OutcomePositive,
+					Probe:    Probe,
+					Message:  "CI test found: pr: 1, context: e2e",
+					Values:   map[string]int{"totalMerged": 1, "totalTested": 1},
+					Location: &finding.Location{Type: 4},
+				},
+			},
+		},
+		{
+			name: "Has 1 CIInfo which has a successful Status.",
+			raw: &checker.RawResults{
+				CITestResults: checker.CITestData{
+					CIInfo: []checker.RevisionCIInfo{
+						{
+							HeadSHA:           "HeadSHA",
+							PullRequestNumber: 1,
+							CheckRuns: []clients.CheckRun{
+								{
+									Status:     "incomplete",
+									Conclusion: "not successful",
+									App:        clients.CheckRunApp{Slug: "e2e"},
+								},
+							},
+							Statuses: []clients.Status{
+								{
+									State:     "success",
+									Context:   CheckCITests,
+									TargetURL: "e2e",
+								},
+							},
+						},
+					},
+				},
+			},
+			findings: []*finding.Finding{
+				{
+					Outcome:  finding.OutcomePositive,
+					Probe:    Probe,
+					Message:  "CI test found: pr: HeadSHA, context: CI-Tests",
+					Values:   map[string]int{"totalMerged": 1, "totalTested": 1},
+					Location: &finding.Location{Type: 4},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			findings, s, err := Run(tt.raw)
+			if !cmp.Equal(tt.err, err, cmpopts.EquateErrors()) {
+				t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(tt.err, err, cmpopts.EquateErrors()))
+			}
+			if err != nil {
+				return
+			}
+			if diff := cmp.Diff(Probe, s); diff != "" {
+				t.Errorf("mismatch (-want +got):\n%s", diff)
+			}
+			if diff := cmp.Diff(len(tt.findings), len(findings)); diff != "" {
+				t.Errorf("mismatch (-want +got):\n%s", diff)
+			}
+			for i := range tt.findings {
+				outcome := &tt.findings[i]
+				f := &findings[i]
+				if diff := cmp.Diff(*outcome, f); diff != "" {
```

Comment:
should we be comparing the other parts of the finding? location?

Reply:
We are checking the whole finding on this line. `*outcome` here a `finding.Finding` from the `tt.findings`.

If I comment out the Location of the `tt.findings` and run the test, it fails:

```golang
	findings: []*finding.Finding{
		{
			Outcome:  finding.OutcomePositive,
			Probe:    Probe,
			Message:  "CI test found: pr: HeadSHA, context: CI-Tests",
			//Location: &finding.Location{Type: 4},
		},
	},
```

from `probes/testsRuninCI`: `go test -run=Test_Run`

```
--- FAIL: Test_Run (0.00s)
    --- FAIL: Test_Run/Has_1_CIInfo_which_has_a_successful_Status. (0.00s)
        impl_test.go:139: mismatch (-want +got):
              &finding.Finding{
                Probe:       "testsRunInCI",
                Outcome:     12,
                Message:     "CI test found: pr: HeadSHA, context: CI-Tests",
            -   Location:    nil,
            +   Location:    &finding.Location{Type: 4},
                Remediation: nil,
                Values:      nil,
              }
FAIL
FAIL    github.com/ossf/scorecard/v4/probes/testsRunInCI        0.025s
FAIL
```

Closing, since I believe it is complete, but please reopen if you think that is not the case.